herddb-core/src/main/java/herddb/sql/CalcitePlanner.java

```
@@ -341,6 +341,84 @@ public TranslatedQuery translate(String defaultTableSpace, String query, List<Ob
+                        joiner.add("Primary key("+ Arrays.stream(t.getPrimaryKey()).collect(Collectors.joining(",")) + ")");
+                    }
+                    sb.append(joiner.toString());
+
+                    ValuesOp values = new ValuesOp(manager.getNodeId(),
+                            new String[]{"tabledef"},
+                            new Column[]{
+                                    column("tabledef", ColumnTypes.STRING)},
+                            Arrays.asList(
+                                    Arrays.asList(
+                                            new ConstantExpression(sb.toString())
+                                    )
+                            )
+                    );
+                    ExecutionPlan executionPlan = ExecutionPlan.simple(
+                            new SQLPlannedOperationStatement(values),
+                            values
+                    );
+                    return new TranslatedQuery(executionPlan, new SQLStatementEvaluationContext(query, parameters));
+                }
+
+            }catch (NullPointerException ex) {
+                ValuesOp values = new ValuesOp(manager.getNodeId(),
+                        new String[]{"error"},
+                        new Column[]{
+                                column("error", ColumnTypes.STRING)},
+                        Arrays.asList(
+                                Arrays.asList(
+                                        new ConstantExpression("Table not found")
+                                )
+                        )
+                );
+                ExecutionPlan executionPlan = ExecutionPlan.simple(
+                        new SQLPlannedOperationStatement(values),
+                        values
+                );
+                return new TranslatedQuery(executionPlan, new SQLStatementEvaluationContext(query, parameters));
+            }
+        }
+        ValuesOp values = new ValuesOp(manager.getNodeId(),
```

Please throw an error, the client must see an hard error

---

Makes sense. Will do.